h/api/search/transform.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+
+"""Functions for transforming data for or from the search index."""
+
+import copy
+
+from h.api import uri
+
+
+def prepare(annotation):
+    """
+    Prepare the given annotation for indexing.
+
+    Scan the passed annotation for any target URIs or document metadata URIs
+    and add normalised versions of these to the document.
+    """
+
+    # FIXME: When this becomes simply part of a search indexing operation, this
+    # should probably not mutate its argument.
+    _normalise_annotation_target_uris(annotation)
+
+
+def render(annotation):
+    """
+    Render an annotation retrieved from search for public display.
+
+    Receives data direct from the search index and reformats it for rendering
+    or display in the public API. At the moment all this does is remove
+    normalised URI fields.
+    """
+    data = copy.deepcopy(annotation)
+
+    _filter_target_normalised_uris(data)
+
+    return data
+
+
+def _normalise_annotation_target_uris(annotation):
+    if 'target' not in annotation:
+        return
+    if not isinstance(annotation['target'], list):
+        return
+    for target in annotation['target']:
+        if not isinstance(target, dict):
+            continue
+        if 'source' not in target:
+            continue
+        if not isinstance(target['source'], basestring):
+            continue
+        target['source_normalised'] = uri.normalise(target['source'])
```

Comment:
Right. I did consider that -- but I think for now it's clearer to keep this separate. Certainly, at some point I would imagine that `hypothesis reindex` is a tool that pulls stuff out of the database, prepares it for search, and then shoves it in elasticsearch. 

When we do that, I'd agree with you that `_source` excludes might be one option for simplifying what we retrieve from the search index. But until that point I think it's going to be confusing to have a `reindex` function that doesn't work. (And updating `reindex` is outside of the the scope of this pull request.)